### PR TITLE
Logout and session management doc

### PR DESCRIPTION
**Pull request by mooreds**
Open questions:

* Do we want to document the `[@helpers.hidden name="metaData.device.description" value="yourdatahere" /]` hack that puts values on the SSO refresh token?
* Is this more than one doc?
* Should we push the terms 'application managed sessions' and 'centralized sessions' into any other places in the doc?

### CODE REVIEW

File: astro/src/content/docs/lifecycle/authenticate-users/logout-session-management.mdx [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+* Richard logs in to ChangeBank with the Chrome browser on his Windows PC. He gets session A.
+* Richard logs in to ChangeBank with the Edge browser on his Windows PC. He gets session B.
+* Richard logs in to the ChangeBank Forum with the Safari browser on his iPhone. He gets session C.
+* Malia logs in to the ChangeBank Forum with the Safari browser on her iPhone. She gets session D.
+
+Even though Richard is on one Windows PC, he gets both session A and session B because he used different browsers.
+
+There are three types of sessions relevant to this guide:
+
+* An **application session** is created after FusionAuth has authenticated a user, the application has completed the token exchange, and the user has been logged in to the application. The application almost always creates a session. How exactly this session works (a cookie, value stored in redis, or a database row) is specific to each web or mobile application.
```
Comment (alex-fusionauth) [inline astro/src/content/docs/lifecycle/authenticate-users/logout-session-management.mdx:63]: Please add `[Rr]edis` to vale

File: astro/src/content/docs/lifecycle/authenticate-users/logout-session-management.mdx [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+  * You could also store the refresh token Id in a HttpOnly cookie instead of in a session.
+* Whenever the ChangeBank or ChangeBank Forum application receives a request, call the [Retrieve Refresh Tokens API](/docs/apis/jwt#retrieve-refresh-tokens), using the stored refresh token Id, to check the status of the refresh token. You can call the REST API directly or using one of [the FusionAuth client libraries](/docs/sdks). If the refresh token doesn't exist or is invalid, deny the user access and invalidate the application session.
+* When a user logs out from either the ChangeBank or ChangeBank Forum application, revoke the refresh token. Use the [Revoke Refresh Tokens API](/docs/apis/jwt#revoke-refresh-tokens). You can revoke all refresh tokens or just a few. The logic around revocation depends on your business needs. See [Flexible Revocation](#flexible-revocation) for more details.
+
+You are using the refresh token to tie the ChangeBank and ChangeBank Forum applications to FusionAuth. Your applications are then confirming session liveness with FusionAuth each time a request is received.
```
Comment (alex-fusionauth) [inline astro/src/content/docs/lifecycle/authenticate-users/logout-session-management.mdx:127]:
```suggestion
You are using the refresh token to tie the ChangeBank and ChangeBank Forum applications to FusionAuth. Your applications are then confirming session validity with FusionAuth each time a request is received.
```

**Review by alex-fusionauth — CHANGES_REQUESTED** [review]
Nice work on this one, wowsers that was a LOT!

File: astro/src/content/docs/lifecycle/authenticate-users/logout-session-management.mdx (hunk 2 above)
Comment (mooreds) [reply astro/src/content/docs/lifecycle/authenticate-users/logout-session-management.mdx]: Didn't accept your suggestion, but did tweak the language. Let me know if it works.

File: astro/src/content/docs/apis/jwt.mdx [hunk 3]
```diff
@@ -287,6 +289,8 @@ Cookie: refresh_token=xRxGGEpVawiUak6He367W3oeOfh+3irw+1G1h1jc; access_token=eyJ
 
 ## Retrieve Refresh Tokens
 
+This can be used to examine a centrally managed session, when the refresh token is being used to model a session. [Learn more about using refresh tokens to model sessions](/docs/lifecycle/authenticate-users/logout-session-management).
```
[the remark this answers is not in the document]
Comment (mooreds) [reply astro/src/content/docs/apis/jwt.mdx]: Didn't accept your suggestion, but did tweak the language. Let me know if it works.

File: astro/src/content/docs/apis/jwt.mdx [hunk 4]
```diff
@@ -327,6 +331,8 @@ Cookie: refresh_token=xRxGGEpVawiUak6He367W3oeOfh+3irw+1G1h1jc; access_token=eyJ
 
 ## Revoke Refresh Tokens
 
+This can be used to revoke a centrally managed session, when the refresh token is being used to model a session. [Learn more about using refresh tokens to model sessions](/docs/lifecycle/authenticate-users/logout-session-management).
```
[the remark this answers is not in the document]
Comment (mooreds) [reply astro/src/content/docs/apis/jwt.mdx:334]: Didn't accept your suggestion, but did tweak the language. Let me know if it works.

File: astro/src/content/docs/apis/jwt.mdx [hunk 5]
```diff
@@ -205,6 +205,8 @@ This API has been available since 1.8.0
 
 ## Refresh a JWT
 
+This can be used to extend a centrally managed session, when the refresh token is being used to model a session. [Learn more about using refresh tokens to model sessions](/docs/lifecycle/authenticate-users/logout-session-management).
```
[the remark this answers is not in the document]
Comment (mooreds) [reply astro/src/content/docs/apis/jwt.mdx]: Didn't accept your suggestion, but did tweak the language. Let me know if it works.

**Comment by mooreds** [comment]
Heya @alex-fusionauth , made some changes you suggested. Also would love your thoughts on the questions in the PR description.

**Comment by alex-fusionauth** [comment]
> Open questions:
> 
> * Do we want to document the `[@helpers.hidden name="metaData.device.description" value="yourdatahere" /]` hack that puts values on the SSO refresh token?
> * Is this more than one doc?
> * Should we push the terms 'application managed sessions' and 'centralized sessions' into any other places in the doc?

* I don't love this, but think it is okay if we are not going to put a full fix in the future. Is that a fix coming from engineering?
* I think the way you have it is solid.
* I really think it is good, unless you are trying for an SEO play.

**Review by alex-fusionauth — APPROVED** [review]

**Comment by mooreds** [comment]
Thanks @alex-fusionauth . re first issue, I chatted with the eng team and I'm not going to document it.

**Comment by alex-fusionauth** [comment]
@mooreds if you want to pull in main we should be good.

**Comment by mooreds** [comment]
On my list for today. Thanks!

**Review by alex-fusionauth — APPROVED** [review]
LGTM!

**Pull request merged**